payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt

```diff
@@ -0,0 +1,224 @@
+import com.stripe.android.paymentsheet.DefaultPrefsRepository
+import com.stripe.android.paymentsheet.GooglePayRepository
+import com.stripe.android.paymentsheet.PaymentOptionContract
+import com.stripe.android.paymentsheet.analytics.DefaultEventReporter
+import com.stripe.android.paymentsheet.analytics.EventReporter
+import com.stripe.android.paymentsheet.analytics.SessionId
+import com.stripe.android.paymentsheet.flowcontroller.ActivityLauncherFactory
+import com.stripe.android.paymentsheet.flowcontroller.DefaultFlowController
+import com.stripe.android.paymentsheet.flowcontroller.DefaultFlowControllerInitializer
+import com.stripe.android.paymentsheet.flowcontroller.FlowControllerInitializer
+import com.stripe.android.paymentsheet.flowcontroller.FlowControllerViewModel
+import com.stripe.android.paymentsheet.model.PaymentIntentClientSecret
+import com.stripe.android.paymentsheet.model.SetupIntentClientSecret
+import dagger.Lazy
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Module
+internal class FlowControllerModule {
+    /**
+     * Provides a non-singleton PaymentConfiguration.
+     *
+     * Needs to be recalculated whenever needed to allow client to set the publishableKey and
+     * stripeAccountId in PaymentConfiguration any time before configuring the FlowController
+     * through configureWithPaymentIntent or configureWithSetupIntent.
+     *
+     * Should always be injected with [Lazy] or [Provider].
+     */
+    @Provides
+    fun providePaymentConfiguration(appContext: Context): PaymentConfiguration {
+        return PaymentConfiguration.getInstance(appContext)
+    }
+
+    // Below are all Singleton instance to be injected into DefaultFlowController
+
+    @Provides
+    @Singleton
+    fun provideFlowControllerInitializer(appContext: Context): FlowControllerInitializer {
+        return DefaultFlowControllerInitializer(
+            prefsRepositoryFactory =
+            { customerId: String, isGooglePayReady: Boolean ->
+                DefaultPrefsRepository(
+                    appContext,
+                    customerId,
+                    { isGooglePayReady },
+                    Dispatchers.IO
+                )
+            },
+            isGooglePayReadySupplier =
+            { environment ->
+                val googlePayRepository = environment?.let {
+                    DefaultGooglePayRepository(
+                        appContext,
+                        it
+                    )
+                } ?: GooglePayRepository.Disabled
+                googlePayRepository.isReady().first()
+            },
+            workContext = Dispatchers.IO
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun provideEventReporter(
+        sessionId: SessionId,
+        appContext: Context
+    ): EventReporter {
+        return DefaultEventReporter(
```

Comment:
This is a constructor that reads from `PaymentConfiguration` itself. We need to use the other one that doesn't instantiate anything internally.

Reply:
nice catch, the ultimate goal would be to fully daggerize all dependencies so that no new instances is created in any surprised way.

Will daggerize `DefaultEventReporter` in a follow up PR